machida3/machida.pony

```diff
@@ -187,10 +188,7 @@ class PyPartitionFunction
         Fail()
       end
 
-      let py_string_p = @PyUnicode_AsUTF8(ps)
-      let py_string_size = @PyUnicode_GetLength(ps)
-
-      let ret = String.copy_cpointer(py_string_p, py_string_size)
+      let ret = String.copy_cstring(@py_bytes_or_unicode_as_char(ps))
```

Comment:
I think we need to stick with `String.copy_cpointer` here. Even though `PyBytes_AsString` returns a null-terminated string, there can be other nulls before the null terminator. `copy_cstring` will copy to the first null, which means we could lose the rest of the value.

I see this pattern used in a lot of places in the code and I think we should fix it in all of them.

@nisanharamati let me know if I'm missing something here.